pkg/crinit/aggregated/aggregated.go

```diff
@@ -0,0 +1,399 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregated
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io"
+
+	"k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/cluster-registry/pkg/apis/clusterregistry/v1alpha1"
+	"k8s.io/cluster-registry/pkg/crinit/util"
+	apiregv1beta1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1"
+	apiregclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var (
+	aggregated_long = `
+	Aggregated initializes an aggregated cluster registry.
+
+	The aggregated cluster registry is hosted inside a Kubernetes
+	cluster and registers its API with the Kubernetes API aggregator.
+	The host cluster must be specified using the --host-cluster-context flag.`
+	aggregated_example = `
+	# Initialize a cluster registry named foo
+	# in the host cluster whose local kubeconfig
+	# context is bar.
+	crinit aggregated foo --host-cluster-context=bar`
+
+	apiServiceGroupPriorityMinimum int32 = 10000
+	apiServiceVersionPriority      int32 = 20
+
+	clusterRegistryAPIService       = v1alpha1.SchemeGroupVersion.Version + "." + v1alpha1.GroupName
```

Comment:
Perhaps you could add comments here as to what all these roles are and what they allow? It would at least help later when someone is writing user documentation.